src/attribution/onINP.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getLoadState} from '../lib/getLoadState.js';
+import {getSelector} from '../lib/getSelector.js';
+import {onINP as unattributedOnINP} from '../onINP.js';
+import {INPMetricWithAttribution, INPReportCallback, INPReportCallbackWithAttribution, ReportOpts} from '../types.js';
+
+
+const attributeINP = (metric: INPMetricWithAttribution): void => {
+  if (metric.entries.length) {
+    const longestEntry = metric.entries.sort((a, b) => {
+      // Sort by: 1) duration (DESC), then 2) processing time (DESC)
+      return b.duration - a.duration || (b.processingEnd - b.processingStart) -
```

Comment:
Oh I like this.  We haven't used the "longest processing time if durations are equal" approach before.

If I recall, metric.entries will be all the entries with the same interactionId, right?  So if we change Event Timing to expose interactionId on more event types, this will "just work" better.  Cool.

Reply:
Yep! I was pretty happy with the simplicity of this approach to be honest :)